sky/backends/cloud_vm_ray_backend.py

```diff
@@ -287,9 +287,10 @@ def add_prologue(self,
                     sys.exit(1)
                 """)
             ]
-        self._code += [
-            f'job_lib.set_status({job_id!r}, job_lib.JobStatus.PENDING)',
-        ]
+        else:
+            self._code += [
+                f'job_lib.set_status({job_id!r}, job_lib.JobStatus.PENDING)',
+            ]
```

Comment:
I am confused. If we don't set the task to `PENDING` after `SETTING_UP`, the job will be waiting for the resources in the `SETTING_UP` status, if the resources is not available on the cluster?

Also, correct me if I am wrong, but I thought you mentioned that we are going to change the job status transition to `INIT` -> `PENDING` -> `SETTING_UP` -> `RUNNING` -> ..., i.e. we will wait for the resource requirements before we start the setup?